src/tools/queryDatasource/validators/validateFilterValues.ts

```diff
@@ -0,0 +1,292 @@
+
+type Filter = z.infer<typeof Filter>;
+
+interface FilterValidationError {
+  field: string;
+  invalidValues: string[];
+  sampleValues: string[];
+  message: string;
+}
+
+/**
+ * Validates SET and MATCH filter values by querying the VDS to check if they exist in the target field.
+ * Returns fuzzy matched sample values from the field if validation fails.
+ */
+export async function validateFilterValues(
+  query: Query,
+  vizqlDataServiceMethods: VizqlDataServiceMethods,
+  datasource: z.infer<typeof Datasource>,
+): Promise<Result<void, FilterValidationError[]>> {
+  if (!query.filters) {
+    return Ok.EMPTY;
+  }
+
+  const validationErrors: FilterValidationError[] = [];
+
+  // Filter for SET and MATCH filters that need validation
+  const filtersToValidate = query.filters.filter(
+    (filter) =>
+      (filter.filterType === 'SET' || filter.filterType === 'MATCH') &&
+      'fieldCaption' in filter.field &&
+      filter.field.fieldCaption,
+  );
+
+  if (filtersToValidate.length === 0) {
+    return Ok.EMPTY;
+  }
+
+  // Validate each filter
+  for (const filter of filtersToValidate) {
+    const fieldCaption = (filter.field as any).fieldCaption;
+
+    try {
+      if (filter.filterType === 'SET') {
+        const result = await validateSetFilter(
+          filter,
+          fieldCaption,
+          vizqlDataServiceMethods,
+          datasource,
+        );
+        if (result.isErr()) {
+          validationErrors.push(result.error);
+        }
+      } else if (filter.filterType === 'MATCH') {
+        const result = await validateMatchFilter(
+          filter,
+          fieldCaption,
+          vizqlDataServiceMethods,
+          datasource,
+        );
+        if (result.isErr()) {
+          validationErrors.push(result.error);
+        }
+      }
+    } catch (error) {
+      // If validation query fails, we'll let the original query proceed
+      // This ensures we don't block valid queries due to validation issues
+      console.warn(`Filter validation failed for field ${fieldCaption}:`, error);
+    }
+  }
+
+  if (validationErrors.length > 0) {
+    return new Err(validationErrors);
+  }
+
+  return Ok.EMPTY;
+}
+
+/**
+ * Validates a SET filter by checking if all values exist in the target field
+ */
+async function validateSetFilter(
+  filter: Filter & { filterType: 'SET' },
+  fieldCaption: string,
+  vizqlDataServiceMethods: VizqlDataServiceMethods,
+  datasource: z.infer<typeof Datasource>,
+): Promise<Result<void, FilterValidationError>> {
+  const filterValues = filter.values.map((v) => String(v));
+
+  // Query to get distinct values from the field
+  const distinctValuesQuery: z.infer<typeof QueryType> = {
+    fields: [
+      {
+        fieldCaption: fieldCaption,
+        fieldAlias: 'DistinctValues',
+      },
+    ],
+  };
+
+  const queryRequest: z.infer<typeof QueryRequest> = {
+    datasource,
+    query: distinctValuesQuery,
+    options: {
+      returnFormat: 'OBJECTS',
+      debug: false,
```

Comment:
Was setting `debug` to false intentional? https://github.com/tableau/tableau-mcp/pull/27